jib-build-plan/src/main/java/com/google/cloud/tools/jib/api/buildplan/ContainerBuildPlan.java

```diff
@@ -422,6 +459,20 @@ public String getOsHint() {
     return osHint;
   }
 
+  /**
+   * Creates and adds the user specified platform to the platforms list if user opts to use
+   * setOsHint and setArchitectureHint methods.Else the method just returns the platforms list set
+   * by the user using addPlatform or setPlatforms methods.
+   *
+   * @return platforms a list of desired platforms.
+   */
+  public List<Platform> getPlatforms() {
+    if (platforms.isEmpty()) {
+      platforms.add(new Platform(osHint, architectureHint));
+    }
+    return new ArrayList<>(platforms);
```

Comment:
We need to think about if this behavior is logical. For example,

- case 1
  1. Someone calls `getPlatforms()` at the beginning. It returns linux/amd64.
  2. they call `addPlatform(linux/arm64)`.
  3. `getPlatforms()` returns linux/am64.

- case 2
  1. Someone calls `setPlatforms( [ linux/amd64 ] )`.
  2. `getPlatforms()` returns linux/amd64.
  3. They call `addPlatform(linux/arm64)`.
  4. `getPlatforms()` returns [ linux/am64, linux/arm64 ].

I think it's a bit counter-intuitive and may become a surprise.

Reply:
We need to think about if this behavior is logical. For example,

- case 1
  1. The user already knows that  linux/amd64 is a default platform so I think this is logical
  2. they call `addPlatform(linux/arm64)`.
  3. `getPlatforms()` returns linux/am64 - This is logical to some extent , if `getPlatforms()` returns [[linux/am64] ,[linux/amd64]] ` that means the user is always forced to build the ` [linux,amd64] ` even though he doesnt want it.

I agree with you that case 1 and case 2 are a bit illogical once you pair them but the current implementation of case 1 makes sense to some extent , we dont want the user to get stuck with the default platform.

Reply:
@chanseokoh What are your thoughts? Should i make case 1 point 3 more like case 2 point 3?

Reply:
How do we see people using getArchitectureHint/getOs now? And how do we see people using getPlatform?

Reply:
I expect no external users using get/setOs/ArchitecureHint as of now, since they have no effect.

Thinking about this, I think `addPlatform()` should work in a consistent manner regardless of whether it's called for the first time or the second time. That is, considering the nature of the method name "add", I think it makes sense to always add a new platform.

At the same time, I'd like to have amd64/linux as the default. We can clearly document in the Javadoc of `addPlatform()` something along the line of saying "note that the build plan starts with amd64/linux as the default platform. If you want to reset the default platform instead of adding a new one, use setPlatforms()" or something like that.

And I remember we had a discussion about handling duplicate values. I think we can take of it easily if we use `Set<Platform>` instead of `List<Platform>`. And this behavior should also be documented. (For an actual `Set` implementation, we want to choose `LinkedHashSet` instead of, e.g., `HashSet` in order to preserve the element order.) But just using `Set` doesn't automatically make it remove duplicates, because Java will always treat one `Platform` instance is different from another instance even if their OS and arch values are same. We need to tell Java that if two different `Platform` instances are essentially "equal" if their internal values are same. To do so, you need to implement `Platform.equals()` and `Platform.hashCode()`. Take a look at the sibling `FileEntry` class.